addons/website/static/src/components/fields/fields.js

```diff
@@ -0,0 +1,59 @@
+/** @odoo-module **/
+
+import {PageDependencies} from '@website/components/dialog/page_properties';
+import {Switch} from '@website/components/switch/switch';
+import AbstractFieldOwl from 'web.AbstractFieldOwl';
+import fieldRegistry from 'web.field_registry_owl';
+
+const {useState} = owl;
+
+class FieldPageUrl extends AbstractFieldOwl {
+    setup() {
+        super.setup();
+
+        this.state = useState({
+            redirectOldUrl: false,
+            url: this.value,
+            redirectType: '301',
+        });
+
+        this.serverUrl = window.location.origin;
+        this.pageUrl = this.value;
+    }
+
+    get enableRedirect() {
+        return this.state.url !== this.pageUrl;
+    }
+
+    onChangeRedirectOldUrl(value) {
+        this.state.redirectOldUrl = value;
+    }
+
+    /**
+     * @override
+     */
+    commitChanges() {
+        if (this.enableRedirect) {
+            this._setValue(this.state.url);
+            if (this.state.redirectOldUrl) {
+                return this.rpc({
+                    model: 'website.rewrite',
+                    method: 'create',
+                    args: [{
+                        'name': this.recordData.name,
+                        'redirect_type': this.state.redirectType,
+                        'url_from': this.pageUrl,
+                        'url_to': this.state.url,
```

Comment:
You're right! It's not the right url (after slugify/unique_path...) so I tweaked the code on create: of redirect to be able to use the same url on page properties :+1:

Reply:
I don't really think it is the correct approach, so right now you are sending the same name to 2 endpoints and tweek the code hoping for the 2 endpoints to process to name to generate the same url (does it even work? The second endpoint will try to generate a uniq url based on the name he received, but since the first endpoint already did that and created the page/url, the second endpont will generate another URL?)

The idea would be to really use the url generated by the create and send it to the redirect create, passing/using the correct URL, not trying to re-generate the same one for convenience

So I would note create that new util method and no context pass in rewrite creation

Reply:
Indeed, this is not the best implementation but it works because `website.rewrite` doesn't create a page...), So I think (since there is no clean way to exchange data between page properties dialog and field component after a "save" ) the best implementation here is to pass the values to be handled in write... what do you think?